common/persistence/sql/sqlplugin/db_metrics.go

```diff
@@ -0,0 +1,54 @@
+package sqlplugin
+
+import (
+	"time"
+
+	"go.temporal.io/server/common/metrics"
+)
+
+type DBMetricsReporter struct {
+	interval time.Duration
+	handle   *DatabaseHandle
+	metrics  metrics.Handler
+	quit     chan struct{}
+}
+
+func withDBMetricReporter(dbKind DbKind, handle *DatabaseHandle) {
```

Comment:
nit: Can you name this method `newDBMetricsReporter` and explicitly call `Start` on it. Just for symmetry with where `Stop` is called.